spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java

```diff
@@ -265,9 +268,27 @@ private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobPar
 					}
 				}
 				if (!existsFlag) {
-					result.add(String.format("%s(%s)=%s", key,
+					String param;
+					if (BOOT3_VERSION.equals(schemaTarget)) {
+						if (JobParameter.ParameterType.LONG.equals(jobParametersMap.get(key).getType())) {
+							type = "java.lang.Long";
```

Comment:
Might be less error-prone to use `Long.class.getSimpleName()`

Reply:
Will do.   FYI it is : `Long.class.getCanonicalName()`

Reply:
In this case, `getCanonicalName() == getSimpleName()` as it is not an array, inner class, nested, etc..